tests/support/client.rs

```diff
@@ -249,36 +294,129 @@ impl hyper::client::connect::Connect for Conn {
+                            return Ok(Async::Ready(RunningIo::Http(io, take_running(running))));
+                        }
+
+                        Some(TlsConfig {
+                            client_config,
+                            name,
+                        }) => {
+                            let future = tokio_rustls::TlsConnector::from(client_config.clone())
+                                .connect(DNSName::as_ref(name), io);
+                            ConnectorFuture::Handshake {
+                                future,
+                                running: running.clone(),
+                            }
+                        }
+                    }
+                }
+
+                ConnectorFuture::Handshake { future, running } => {
+                    let io = try_ready!(future.poll());
+                    return Ok(Async::Ready(RunningIo::Https(io, take_running(running))));
+                }
+            }
+        }
+    }
+}
+
+enum RunningIo {
+    Http(TcpStream, oneshot::Sender<()>),
+    Https(TlsStream<TcpStream, ClientSession>, oneshot::Sender<()>),
+}
+
+impl Read for RunningIo {
+    #[inline]
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.inner.read(buf)
+        match *self {
+            RunningIo::Http(ref mut s, _) => s.read(buf),
+            RunningIo::Https(ref mut s, _) => s.read(buf),
+        }
     }
 }
 
-impl io::Write for RunningIo {
+impl Write for RunningIo {
+    #[inline]
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        self.inner.write(buf)
+        match *self {
+            RunningIo::Http(ref mut s, _) => s.write(buf),
+            RunningIo::Https(ref mut s, _) => s.write(buf),
+        }
     }
 
+    #[inline]
     fn flush(&mut self) -> io::Result<()> {
-        self.inner.flush()
+        match *self {
+            RunningIo::Http(ref mut s, _) => s.flush(),
+            RunningIo::Https(ref mut s, _) => s.flush(),
+        }
     }
 }
 
-impl AsyncRead for RunningIo {}
+impl AsyncRead for RunningIo {
+    unsafe fn prepare_uninitialized_buffer(&self, buf: &mut [u8]) -> bool {
+        match *self {
+            RunningIo::Http(ref s, _) => s.prepare_uninitialized_buffer(buf),
+            RunningIo::Https(ref s, _) => s.prepare_uninitialized_buffer(buf),
+        }
+    }
+
+    fn read_buf<B: BufMut>(&mut self, buf: &mut B) -> Poll<usize, io::Error> {
+        match *self {
+            RunningIo::Http(ref mut s, _) => s.read_buf(buf),
+            RunningIo::Https(ref mut s, _) => s.read_buf(buf),
+        }
+    }
+}
 
 impl AsyncWrite for RunningIo {
     fn shutdown(&mut self) -> Poll<(), io::Error> {
-        AsyncWrite::shutdown(&mut self.inner)
+        match *self {
+            RunningIo::Http(ref mut s, _) => s.shutdown(),
+            RunningIo::Https(ref mut s, _) => s.shutdown(),
+        }
+    }
+
+    fn write_buf<B: Buf>(&mut self, buf: &mut B) -> Poll<usize, io::Error> {
```

Comment:
I have the same comment for this method about `AsyncRead`. We should be fine not adding this, but let me know if there were future intentions for it.